src/soong_module.rs

```diff
@@ -70,6 +70,30 @@ impl SoongNamedProp {
         self.prop.clone()
     }
 
+    // A helper method to check whether the type of `prop` is `VecStr` and
+    // whether any string `s` in `prop` has f(s) returning true.
+    // The type F must implement the Fn trait, which is the trait for closures
+    // that take an immutable reference to &str and return a bool.
```

Comment:
The function interface is clear here. The comment is not adding any information that is not already in the interface.